libobs/obs-encoder.h

```diff
@@ -120,6 +123,18 @@ struct obs_encoder_roi {
 	float priority;
 };
 
+struct gs_texture;
+
+/** Encoder input texture */
+struct encoder_texture {
+	/** Texture format and size */
+	struct video_scale_info info;
+	/** Shared texture handle, only set on Windows */
+	uint32_t handle;
+	/** Textures, NULL-terminated */
+	struct gs_texture *tex[5];
```

Comment:
Also not really sure about this, Why does it need to be NULL terminates? The currently in use texture format should determine the number of textures in use. And at most this would then be 4 (Y + U + V + Alpha).

Reply:
The 5th element was an earlier discussion to just be safe and include a terminal, i dont feel strongly.

As this is OBS's api we can decide to always use multiple textures, one per plane. In theory (in reality on vulkan) some platforms only do planar as multiple views on a single alloc, but we can cross that bridge when we get to it.